### PR TITLE
Deal with Github repo redirects 

### PR DESCRIPTION
**Pull request by orta**
We were getting crashes that stopped the github metrics grabbing caused by redirects. It was difficult to run just one metric, so I added a rake task for it.

### CODE REVIEW

File: lib/metrics/github.rb [hunk 1]
```diff
@@ -78,8 +84,7 @@ def handle_update_error(e, pod)
       when /404 Not Found/
         not_found(pod)
       when /403 API rate limit exceeded/
-        headers = ::Github::Response::Headers.new response_headers: e.http_headers
-
+        headers = ::Github::Response::Header.new response_headers: e.http_headers
```
Comment (orta) [inline lib/metrics/github.rb]: There is no `Headers` class only [Header](https://github.com/peter-murach/github/blob/master/lib/github_api/response/header.rb#L7), this was being masked by the `raise` in `updater.rb`

File: Rakefile [hunk 2]
```diff
@@ -41,6 +41,11 @@ begin
     exec 'bundle exec kicker -c'
   end
 
+  desc 'Update metrics for a pod'
+    task :update, [:name] do |t, args|
+      exec "bundle exec ruby updater.rb #{ args[:name] }"
```
Comment (segiddins) [inline Rakefile]: Missing an end

File: apiary.apib [hunk 3]
```diff
@@ -49,7 +49,6 @@ Metrics is an API to view metrics for Pods.
 + closed_pull_requests: 11 (number)
 
 ### CocoaDoc Metrics
-+ download_size: 464 (number)
```
Comment (segiddins) [inline apiary.apib]: Install size now?

File: lib/metrics/github.rb [hunk 4]
```diff
@@ -91,6 +96,22 @@ def check_rate_limiting(response)
       raise build_rate_limit_error(response.headers) if remaining.nil? || remaining.to_i <= 0
     end
 
+    # Support GitHub's repo redirection feature
+    #
+    def check_repo_redirection(client, repo)
+      if repo.respond_to?(:message) && repo.message == "Moved Permanently"
+          redirect_details = REST.get(repo.url).body
```
Comment (segiddins) [inline lib/metrics/github.rb]: Why not check the response status code?

Comment (orta) [inline lib/metrics/github.rb]: Yeah! I found a `redirect?`function, so it uses this instead

File: Rakefile (hunk 2 above)
Comment (orta) [inline Rakefile]: thanks :dancers:

File: apiary.apib (hunk 3 above)
Comment (orta) [inline apiary.apib]: yeah, I went and updated it with all the new metrics, I don't know the format but it fools similar to what we had before

**Comment by orta** [comment]
doing it :green_apple:

File: apiary.apib [hunk 5]
```diff
@@ -22,6 +22,7 @@ Metrics is an API to view metrics for Pods.
 + Attributes
     + github (GitHub Metrics)
     + cocoadocs (CocoaDoc Metrics)
+    + stats (CocoaPods Stats Metrics)
```
Comment (kylef) [inline apiary.apib]: `http://metrics.cocoapods.org/api/v1/status` has github and cocoadocs, should it also have stats now?

File: lib/metrics/github.rb [hunk 6]
```diff
@@ -18,13 +20,16 @@ def initialize(limit, resets_at)
...
       @user, @repo = parse url
-
       params = {
         :user => @user,
         :repo => @repo
       }
+      github_client params
+    end
+
+    def github_client params
```
Comment (kylef) [inline lib/metrics/github.rb]: Missing round brackets around parameters

**Pull request merged**